lib/ProductOpener/Images.pm

```diff
@@ -1349,4 +1349,132 @@ HTML
 	return $html;
 }
 
+
+sub extract_text_from_image($$$$) {
+
+	my $product_ref = shift;
+	my $id = shift;
+	my $field = shift;
+	my $ocr_engine = shift;
+
+	delete $product_ref->{$field};
+
+	my $path = product_path($product_ref->{code});
+	my $status = 1;
+
+	my $filename = '';
+
+	my $lc = $product_ref->{lc};
+
+	if ($id =~ /_(\w\w)$/) {
+		$lc = $1;
+	}
+
+	my $size = 'full';
+	if ((defined $product_ref->{images}) and (defined $product_ref->{images}{$id})
+		and (defined $product_ref->{images}{$id}{sizes}) and (defined $product_ref->{images}{$id}{sizes}{$size})) {
+		$filename = $id . '.' . $product_ref->{images}{$id}{rev} ;
+	}
+
+	my $image = "$www_root/images/products/$path/$filename.full.jpg";
+	my $image_url = format_subdomain('static') . "/images/products/$path/$filename.full.jpg";
+
+	my $text;
+
+	$log->debug("extracting text from image", { id => $id, ocr_engine => $ocr_engine }) if $log->is_debug();
+
+	if ($ocr_engine eq 'tesseract') {
+
+		my $lan;
+
+		if (defined $ProductOpener::Config::tesseract_ocr_available_languages{$lc}) {
+			$lan = $ProductOpener::Config::tesseract_ocr_available_languages{$lc};
+		}
+		elsif (defined $ProductOpener::Config::tesseract_ocr_available_languages{$product_ref->{lc}}) {
+			$lan = $ProductOpener::Config::tesseract_ocr_available_languages{$product_ref->{lc}};
+		}
+		elsif (defined $ProductOpener::Config::tesseract_ocr_available_languages{en}) {
+			$lan = $ProductOpener::Config::tesseract_ocr_available_languages{en};
+		}
+
+		$log->debug("extracting text with tesseract", { lc => $lc, lan => $lan, id => $id, image => $image }) if $log->is_debug();
+
+		if (defined $lan) {
+			$text =  decode utf8=>get_ocr($image,undef,$lan);
+
+			if ((defined $text) and ($text ne '')) {
+				$product_ref->{$field} = $text;
+				$status = 0;
+			}
+		}
+		else {
+			$log->warn("no available tesseract dictionary", { lc => $lc, lan => $lan, id => $id }) if $log->is_warn();
+		}
+
+	}
+	elsif ($ocr_engine eq 'google_cloud_vision') {
+
+		my $url = "https://alpha-vision.googleapis.com/v1/images:annotate?key=" . $ProductOpener::Config::google_cloud_vision_api_key;
```

Comment:
Why `alpha-vision`? The official docs don't mention that subdomain: https://cloud.google.com/vision/docs/reference/rest/

Reply:
We're using the alpha endpoint because I have early and free access. Behaves like the regular endpoint, with more features.

Reply:
Gotcha!